src/world/Spell/SpellAuras.cpp

```diff
@@ -955,15 +955,12 @@ void Aura::Remove()
         caster->removeSingleTargetGuidForAura(GetSpellInfo()->getId());
 
     /* Remove aurastates */
-    uint32 flag = 0;
     if (m_spellInfo->getMechanicsType() == MECHANIC_ENRAGED && !--m_target->asc_enraged)
-        flag |= AURASTATE_FLAG_ENRAGED;
+        m_target->removeAuraState(AURASTATE_FLAG_ENRAGED);
     else if (m_spellInfo->getMechanicsType() == MECHANIC_BLEEDING && !--m_target->asc_bleed)
-        flag |= AURASTATE_FLAG_BLEED;
+        m_target->removeAuraState(AURASTATE_FLAG_BLEED);
     if (m_spellInfo->custom_BGR_one_buff_on_target & SPELL_TYPE_SEAL && !--m_target->asc_seal)
-        flag |= AURASTATE_FLAG_JUDGEMENT;
-    if (flag != 0)
```

Comment:
Please explain the removal of this block.

Reply:
Before this change aurastates were handled in bitmasks. The block is just made "decimal-friendly".